dissect/target/plugins/os/unix/log/lastlog.py

```diff
@@ -39,27 +66,27 @@ def lastlog(self):
             - https://www.tutorialspoint.com/unix_commands/lastlog.htm
         """
         try:
-            wtmp = self.target.fs.open("/var/log/lastlog")
+            lastlog = self.target.fs.open("/var/log/lastlog")
         except FileNotFoundError:
             return
 
-        log = UtmpFile(wtmp)
+        users = dict()
+        for user in self.target.users():
+            users[user.uid] = user.name
+
+        log = LastLogFile(lastlog)
 
-        for entry in log:
+        for idx, entry in enumerate(log):
 
-            if entry.ut_type in utmp.Type.reverse:
-                r_type = utmp.Type.reverse[entry.ut_type]
-            else:
-                r_type = None
+            # if ts=0 the uid has never logged in before
+            if entry.ut_host.decode().strip("\x00") == "" or entry.ll_time.tv_sec == 0:
```

Comment:
```suggestion
            if entry.ut_host.strip(b"\x00") == b"" or entry.ll_time.tv_sec == 0:
```
Saves a few cycles decoding and prevents possible decoding errors.